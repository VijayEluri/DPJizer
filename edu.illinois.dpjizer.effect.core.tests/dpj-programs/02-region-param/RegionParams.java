/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package regionparam.regionparams;

/**
 * 
 * @author Mohsen Vakilian
 * @author Danny Dig
 *
 */
class C1<region R1> {
	region r1;
	int x in R1;
	
	C1() {}
	
	void m() {
		x = 2;
		new C2<r1>().m();
	}
}

class C2<region R2> {
	region r2;
	int x in R2;
	
	C2() {}
	
	void m() {
		x = 2;
		new C1<r2>().m();
	}
}

