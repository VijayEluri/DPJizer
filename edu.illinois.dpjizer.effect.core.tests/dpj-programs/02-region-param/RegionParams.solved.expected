regionparam.regionparams.C1.m writes R1, Root : regionparam.regionparams.C1.r1, Root : regionparam.regionparams.C2.r2
regionparam.regionparams.C2.m writes R2, Root : regionparam.regionparams.C1.r1, Root : regionparam.regionparams.C2.r2
